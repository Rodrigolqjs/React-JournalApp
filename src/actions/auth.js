// import { FirebaseError } from 'firebase/app';
import { createUserWithEmailAndPassword, getAuth, signInWithEmailAndPassword, signInWithPopup, signOut, updateProfile } from 'firebase/auth'
import { googleAuthProvider } from '../firebase/firebase-config';
import {types} from '../types/types';
import { finishLoading, startLoading } from './ui';
import Swal from 'sweetalert2'
import { notesLogout } from './notes';
 
export const startLoginEmailPassword = (email, password) =>{
    return (dispatch) => {

        // console.log(email, password);
            dispatch( startLoading() );

            const auth = getAuth();

            return signInWithEmailAndPassword(auth, email, password)

            .then(({ user }) => {
                // console.log(user.uid, user.displayName);
                dispatch( login(user.uid, user.displayName) )
                dispatch( finishLoading() );
                
            })

            .catch( e => {
                console.log(e);
                dispatch( finishLoading() )
                Swal.fire( 'Fail', e.message, 'error' );
            } );
    }
}

export const startRegisterWithEmailPasswordName = ( email, password, name ) => {

    return (dispatch) => {
        
        const auth = getAuth();
            createUserWithEmailAndPassword(auth, email, password)
            .then( async({ user }) => {
 
                await updateProfile( user, { displayName: name });
                
            })

            .catch( e => {
                console.log(e);
                Swal.fire( 'Fail', e.message, 'error' );
            })
    }
}
 
export const startGoogleLogin = () =>{
    return (dispatch) => {
        const auth = getAuth();
        signInWithPopup(auth, googleAuthProvider)
            .then(({user}) =>{
                dispatch(login(user.uid, user.displayName))
            });
    }
}
 
export const login = (uid, displayName) =>(
    {
        type:types.login,
        payload: {
            uid,
            displayName
        }
    }
);

export const startLogout = () => {
    return async( dispatch ) => {
        const auth = getAuth();
        await signOut(auth)

        dispatch( logout() )
        dispatch(notesLogout())
    }
}

export const logout = () => ({
    type: types.logout
})

