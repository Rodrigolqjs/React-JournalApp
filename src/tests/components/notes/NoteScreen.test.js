import React from 'react';
import configureStore from 'redux-mock-store' //ES6 modules
import thunk from "redux-thunk";
import '@testing-library/jest-dom'
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/notes/NoteScreen';

jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn()
}))
// jest.mock('../../../../actions/notes', () => ({
//     startNewNote: jest.fn()
// }))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const initState = {
    auth: {
        uid: 1,
        name: 'Rodrigo'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: 'abc',
            title: 'hola,',
            body: 'mundo',


        },
        notes: []
    }

}

let store = mockStore(initState)
store.dispatch = jest.fn();

const wrapper = mount( 
    <Provider store={ store }>
        {/* <MemoryRouter> */}
            <NoteScreen /> 
        {/* </MemoryRouter> */}
    </Provider>
)



describe('Pruebas en <NoteScreen />', () => {
    
    test('debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    })

    test('debe de disparar el active note', () => {
        
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'Hola de nuevo'
            }
        });

        expect( activeNote ).toHaveBeenLastCalledWith( 
            'abc',
            {
                body: 'mundo',
                title: 'Hola de nuevo',
                id: 'abc',
            }
        );

    })
    
    
})
