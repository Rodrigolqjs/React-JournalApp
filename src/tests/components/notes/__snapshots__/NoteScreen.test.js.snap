// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NoteScreen /> debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      28 de agosto 2020
    </span>
    <input
      id="fileSelector"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="false"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Este es el titulo"
      type="text"
      value="hola,"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="algo asombroso aqui"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
